### PR TITLE
[CST] Updates to prep for lighthouse migration

### PR DESCRIPTION
**Pull request by jerekshoe**
## Summary
This PRs' purpose is to help lay out the groundwork for the upcoming migration from EVSS to Lighthouse. It cleans up some of the code and adds two new things:
* Keys passed to `recordEvent` have copies with `-` replaced with `_` to prep for GA4 migration
* A recently closed claim is injected into the mocked list of claims used for local development so that we can ensure that that feature continues to work

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#50036
- *Link to ticket created in va.gov-team repo*
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done
Tests still pass; There shouldn't be any functional updates

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/claims-status/components/ClosedClaimMessage.jsx [hunk 1]
```diff
@@ -63,6 +71,20 @@ export default function ClosedClaimMessage({ claims, onClose }) {
 
       return c;
     });
+};
+
+const getCloseDate = claim => claim.attributes.phaseChangeDate;
```
Comment (kjduensing) [inline src/applications/claims-status/components/ClosedClaimMessage.jsx:76]: Is the `phaseChangeDate` always the close date? I was working under the assumption that the `phaseChangeDate` is the date the claim changed to _any_ phase.

Comment (jerekshoe) [reply src/applications/claims-status/components/ClosedClaimMessage.jsx:76]: So the way that the vets-api endpoint works currently is there is no attribute to store the `closeDate`. The way that the frontend knows that the claim is closed is by looking at the `open` attribute. It this attribute is set to **false**, then we look at the `phaseChangeDate` to see when the last phase change occurred. The list of claims is filtered down to only include claims where the `open` attribute is set to **false** and where the `phaseChangeDate` is within the last 30 days by the `getRecentlyClosedClaims` method that is called earlier in the component, so in theory `phaseChangeDate` should always represent the close date. The new API will have an explicit property `closeDate` to indicate whether the date the claim was closed on

Comment (kjduensing) [reply src/applications/claims-status/components/ClosedClaimMessage.jsx:76]: This makes sense, thank you

**Review by kjduensing — APPROVED** [review]
Looks great 🦖

**Pull request merged**